Orange/widgets/visualize/owscatterplot.py

```diff
@@ -124,8 +124,8 @@ class Outputs:
     auto_sample = Setting(True)
     toolbar_selection = Setting(0)
 
-    attr_x = ContextSetting(None)
-    attr_y = ContextSetting(None)
+    attr_x = ContextSetting(None, required=ContextSetting.REQUIRED)
+    attr_y = ContextSetting(None, required=ContextSetting.REQUIRED)
```

Comment:
I'm puzzled here. I thought that the default for `required` was `ContextSetting.REQUIRED`. I implemented widgets under this assumption. Now I see it's actually not. So, for instance, Sieve diagram, which does not mark its attributes as required, can get a context without these attributes? It reuses invalid contexts and does not crash just because it can keep the attributes from before the context was opened? Could you take a look?

We may want to comb through all widgets and check the settings. I would make `REQUIRED` the default, and only use `OPTIONAL` in the few places like colors and shapes in the scatter plot. Not necessarily in this pull request; we can open this as an issue.

Reply:
Until this PR is merged, everything was marked as OPTIONAL, but treated as required.
I will amend the last commit and change the default to required.